linode/networkingip/framework_resource.go

```diff
@@ -0,0 +1,220 @@
+package networkingip
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/linode/linodego"
+	"github.com/linode/terraform-provider-linode/v2/linode/helper"
+)
+
+func NewResource() resource.Resource {
+	return &Resource{
+		BaseResource: helper.NewBaseResource(
+			helper.BaseResourceConfig{
+				Name:   "linode_networking_ip",
+				IDType: types.StringType,
+				Schema: &frameworkResourceSchema,
+			},
+		),
+	}
+}
+
+type Resource struct {
+	helper.BaseResource
+}
+
+func (r *Resource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	tflog.Debug(ctx, "Create linode_networking_ip")
+	var plan NetworkingIPModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	client := r.Meta.Client
+
+	createOpts := linodego.AllocateReserveIPOptions{
+		Type:   plan.Type.ValueString(),
+		Public: plan.Public.ValueBool(),
+	}
+
+	if !plan.LinodeID.IsNull() {
+		createOpts.LinodeID = int(plan.LinodeID.ValueInt64())
+	}
+	if !plan.Reserved.IsNull() {
+		createOpts.Reserved = plan.Reserved.ValueBool()
+	}
+	if !plan.Region.IsNull() {
+		createOpts.Region = plan.Region.ValueString()
+	}
+
+	ip, err := client.AllocateReserveIP(ctx, createOpts)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating IP Address",
+			fmt.Sprintf("Could not create IP address: %s", err),
+		)
+		return
+	}
+
+	plan.FlattenIPAddress(ctx, ip, false)
+	resp.Diagnostics.Append(resp.State.Set(ctx, plan)...)
+}
+
+func (r *Resource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	tflog.Debug(ctx, "Read linode_networking_ip")
+	var state NetworkingIPModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	client := r.Meta.Client
+
+	ip, err := client.GetIPAddress(ctx, state.ID.ValueString())
+	if err != nil {
+		if lerr, ok := err.(*linodego.Error); ok && lerr.Code == 404 {
+			resp.State.RemoveResource(ctx)
+			return
+		}
```

Comment:
It looks like private ephemeral IPs are always hitting this condition (404) on post-creation reads, leading to the resource being recreated on the next apply. Do they need to be accessed through a different endpoint?

Reply:
That is true. The GetIPAddress() method only works with public IP Addresses. The resource also only supports public IP's for now. So I guess it is ok to leave it as is for now?

Reply:
The behavior is correct. As confirmed in this [Slack thread](https://linode.slack.com/archives/CDXF1608G/p1734362729544919), there is no API available to fetch private IP addresses. To address this issue, we have two possible workarounds:

1. Use the /networking/ips API to list all addresses and manually filter out the specific private IP address.
2. Log a warning to indicate that the resource currently only supports public IPs, with the possibility of adding support for private IPs in the future.